yarpl/include/yarpl/observable/Observables.h

```diff
@@ -47,6 +47,21 @@ class Observables {
     return Observable<T>::create(std::move(lambda));
   }
 
+  // this will generate an observable which can be subscribed to only once
+  template <typename T>
+  static Reference<Observable<T>> justOnce(T value) {
+    auto lambda = [value = std::move(value), used = false](Reference<Observer<T>> observer) mutable {
+      CHECK(!used) << "justOnce value was already used";
```

Comment:
I think this should resolve with an exception in the other case, crashing doesn't seem correct.